node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -25,6 +27,50 @@ pub enum TxStatus {
     Succeeded(TransactionBlock),
 }
 
+impl FromStr for TxStatus {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Pending" => Ok(TxStatus::Pending),
+            "Failed" => Ok(TxStatus::Failed),
```

Comment:
Add a TODO mentioning the new card we talked about, for failed txs. 

Also remember that the new table should store also some information which will allow us to tell the failures apart by their ranking. We will definitely want to know as much as we can read out regarding the possible cause. (I'm writing it here because I was imagining it even in this table and as I see you did not go into this I just want to emphasize how important it is. Here I would've been expecting FailureMark<AndItsIdentification>, but we moved away from there and now the plans are different.)